packages/machine/test/integration/test-io-provider.ts

```diff
@@ -34,70 +31,30 @@ export class TestIOProvider {
       serializedMessage
     ) as cf.legacy.node.ClientActionMessage;
 
-    let done = false;
-    const executedListeners = [] as number[];
-    let count = 0;
-
-    // Invoke all listeners waiting for a response to resolve their promise
-    this.listeners.forEach(listener => {
-      if (
-        listener.appId === message.appId ||
-        (!listener.appId && listener.multisig === message.multisigAddress)
-      ) {
-        listener.method(message);
-        done = true;
-        executedListeners.push(count);
-        count += 1;
-      }
-    });
-
-    // Now remove all listeners we just invoked
-    executedListeners.forEach(index => this.listeners.splice(index, 1));
-
-    // initiate ack side if needed
-    if (message.seq === 1) {
+    if (this.waitForIoContinuation !== undefined) {
+      const k = this.waitForIoContinuation;
+      this.waitForIoContinuation = undefined;
```

Comment:
It's not immediately clear what the intent here is.

Why are we reassigning `undefined` to this?

Reply:
I think the idea is that you had a continuation ready to be used on some data when the time was right, and this code snippet is only executed "when the time is right" and so it executes it and gets rid of the continuation

Reply:
tbh I think it's a cool use of continuations

Reply:
@cf19drofxots is right, the reassignment is not actually needed

Reply:
ah actually it is needed, because of the check here https://github.com/counterfactual/monorepo/pull/247/files#diff-dce421262ecb0310b0ecc493a5bd9fbeR33